miniapp/src/test/java/com/rakuten/tech/mobile/miniapp/display/RealMiniAppDisplayTest.kt

```diff
@@ -0,0 +1,70 @@
+package com.rakuten.tech.mobile.miniapp.display
+
+import android.content.Context
+import android.view.ViewGroup
+import android.webkit.WebView
+import androidx.test.core.app.ApplicationProvider.getApplicationContext
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.rakuten.tech.mobile.miniapp.TEST_BASE_PATH
+import kotlinx.coroutines.test.runBlockingTest
+import org.amshove.kluent.*
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import kotlin.test.assertTrue
+
+@RunWith(AndroidJUnit4::class)
+class RealMiniAppDisplayTest {
+
+    lateinit var context: Context
+
+    @Before
+    fun setup() {
+        context = getApplicationContext()
+    }
+
+    @Test
+    fun `for a given basePath, obtainView returns corresponding view to the caller`() =
+        runBlockingTest {
+            val display = Displayer(context).createMiniAppDisplay(basePath = TEST_BASE_PATH)
+            val miniAppView = display.getMiniAppView()
+            miniAppView shouldNotHaveTheSameClassAs WebView(context)
+            if (miniAppView is WebView) {
+                miniAppView.url shouldContain TEST_BASE_PATH
+            }
+        }
+
+    @Test
+    fun `for a given basePath, obtainView should not return WebView to the caller`() =
+        runBlockingTest {
+            val display = Displayer(context).createMiniAppDisplay(basePath = TEST_BASE_PATH)
+            val miniAppView = display.getMiniAppView()
+            miniAppView shouldNotHaveTheSameClassAs WebView::class
+        }
+
+    @Test
+    fun `when getLoadUrl is called then a formed path is returned from basePath for loading`() {
+        val miniAppWindow = RealMiniAppDisplay(context, basePath = TEST_BASE_PATH)
+        miniAppWindow.getLoadUrl() shouldBeEqualTo "file://dummy/index.html"
```

Comment:
As I mentioned, I think we should test this by calling `WebView#url` instead of exposing `getLoadUrl`. Checking this exposed method doesn't guarantee that the webview is actually using that method.

I know that we are supposed to be returning a generic `View` and not a `WebView`, but I think it's okay to treat it as a `WebView` in the tests for `RealMiniAppDisplay`.

Reply:
This is only attempting to check if the path is correctly formed when the `basePath` is supplied into the constructor.